tracer/src/Datadog.Trace.Tools.dd_dotnet/Checks/ProcessBasicCheck.cs

```diff
@@ -0,0 +1,463 @@
+
+            string corProfilerKey = runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER" : "COR_PROFILER";
+
+            process.EnvironmentVariables.TryGetValue(corProfilerKey, out var corProfiler);
+
+            if (corProfiler != Utils.Profilerid)
+            {
+                Utils.WriteWarning(WrongEnvironmentVariableFormat(corProfilerKey, Utils.Profilerid, corProfiler));
+                ok = false;
+            }
+
+            string corEnableKey = runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_ENABLE_PROFILING" : "COR_ENABLE_PROFILING";
+
+            process.EnvironmentVariables.TryGetValue(corEnableKey, out var corEnable);
+
+            if (corEnable != "1")
+            {
+                Utils.WriteError(WrongEnvironmentVariableFormat(corEnableKey, "1", corEnable));
+                ok = false;
+            }
+
+            ok &= CheckProfilerPath(process, runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER_PATH" : "COR_PROFILER_PATH", requiredOnLinux: true);
+            ok &= CheckProfilerPath(process, runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER_PATH_32" : "COR_PROFILER_PATH_32", requiredOnLinux: false);
+            ok &= CheckProfilerPath(process, runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER_PATH_64" : "COR_PROFILER_PATH_64", requiredOnLinux: false);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (!CheckRegistry(registryService, nativeTracerVersion))
+                {
+                    ok = false;
+                }
+            }
+
+            if (process.EnvironmentVariables.TryGetValue("DD_TRACE_ENABLED", out var traceEnabledValue))
+            {
+                if (!ParseBooleanConfigurationValue(traceEnabledValue))
+                {
+                    Utils.WriteError(TracerNotEnabled(traceEnabledValue));
+                }
+            }
+
+            return ok;
+        }
+
+        internal static bool CheckContinousProfiler(ProcessInfo process, string? loaderModule)
+        {
+            bool ok = true;
+
+            var continuousProfilerModule = FindContinousProfilerModule(process);
+
+            if (continuousProfilerModule == null)
+            {
+                Utils.WriteWarning(ContinuousProfilerNotLoaded);
+                ok = false;
+            }
+
+            if (loaderModule == null)
+            {
+                Utils.WriteError(ContinuousProfilerWithoutLoader);
+                ok = false;
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                process.EnvironmentVariables.TryGetValue("LD_PRELOAD", out var ldPreload);
+
+                if (ldPreload == null)
+                {
+                    Utils.WriteError(LdPreloadNotSet);
+                    ok = false;
+                }
+                else
+                {
+                    if (Path.GetFileName(ldPreload) != "Datadog.Linux.ApiWrapper.x64.so")
```

Comment:
For later, it's possible to have more than one library in the LD_PRELOAD env var (ex the injection library) and also in some specific case (container with capabilities), we have to check if the file in `/etc/ld.so.preload` 
(mainly a note for me :) )